sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -100,39 +100,75 @@ private class CosmosRowConverter(
 
     def fromRowToObjectNode(row: Row): ObjectNode = {
 
-        if (row.schema.contains(StructField(CosmosTableSchemaInferrer.RawJsonBodyAttributeName, StringType))){
-            // Special case when the reader read the rawJson
-            val rawJson = row.getAs[String](CosmosTableSchemaInferrer.RawJsonBodyAttributeName)
-            objectMapper.readTree(rawJson).asInstanceOf[ObjectNode]
-        }
-        else {
-            val objectNode: ObjectNode = objectMapper.createObjectNode()
-            row.schema.fields.zipWithIndex.foreach({
-              case (field, i) =>
-                field.dataType match {
-                  case _: NullType => putNullConditionally(objectNode, field.name)
-                  case _ if row.isNullAt(i) => putNullConditionally(objectNode, field.name)
-                  case _ =>
-                    val nodeOpt = convertSparkDataTypeToJsonNode(field.dataType, row.get(i))
-                    if (nodeOpt.isDefined) {
-                      objectNode.set(field.name, nodeOpt.get)
-                    }
+      val rawBodyFieldName = if (row.schema.names.contains(CosmosTableSchemaInferrer.RawJsonBodyAttributeName) &&
+        row.schema.apply(CosmosTableSchemaInferrer.RawJsonBodyAttributeName).dataType.isInstanceOf[StringType]) {
+        Some(CosmosTableSchemaInferrer.RawJsonBodyAttributeName)
+      } else if (row.schema.names.contains(CosmosTableSchemaInferrer.OriginRawJsonBodyAttributeName) &&
+        row.schema.apply(CosmosTableSchemaInferrer.OriginRawJsonBodyAttributeName).dataType.isInstanceOf[StringType]) {
+        Some(CosmosTableSchemaInferrer.OriginRawJsonBodyAttributeName)
+      } else {
+        None
+      }
+
+      if (rawBodyFieldName.isDefined){
+
+        val rawBodyFieldIndex = row.schema.fieldIndex(rawBodyFieldName.get)
+        // Special case when the reader read the rawJson
+        val rawJson = row.getAs[String](rawBodyFieldName.get)
```

Comment:
These 2 lines could be moved inside the new shared method, and avoid the boolean parameter (keepOriginProperties).

if the new method would be called:

`private def generateRawJsonNode(row, rawBodyFieldName): ObjectNode`

If could make all the logical switches inside it needs, and would reduce the duplicity and having the `keepOriginProperties` logic inside, instead of spreading it around?

Reply:
Done (next iteration)